internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -61,54 +62,57 @@ func resourceComputeCluster() *pluginsdk.Resource {
 				ValidateFunc: validation.StringInSlice([]string{string(machinelearningservices.VMPriorityDedicated), string(machinelearningservices.VMPriorityLowPriority)}, false),
 			},
 
-			"identity": {
+			"identity": SystemAssignedUserAssigned{}.Schema(),
+
+			"scale_settings": {
 				Type:     pluginsdk.TypeList,
 				Required: true,
 				ForceNew: true,
 				MaxItems: 1,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
-						"type": {
-							Type:     pluginsdk.TypeString,
+						"max_node_count": {
+							Type:     pluginsdk.TypeInt,
 							Required: true,
 							ForceNew: true,
-							ValidateFunc: validation.StringInSlice([]string{
-								string(machinelearningservices.ResourceIdentityTypeSystemAssigned),
-							}, false),
 						},
-						"principal_id": {
-							Type:     pluginsdk.TypeString,
-							Computed: true,
+						"min_node_count": {
+							Type:     pluginsdk.TypeInt,
+							Required: true,
+							ForceNew: true,
 						},
-						"tenant_id": {
+						"scale_down_nodes_after_idle_duration": {
```

Comment:
Could you please elaborate why we are unable to change the name?

Reply:
Once a property has been released to users we do not change it until a major version change - otherwise we would break users terraform configurations during a .x release which we do not do.

additionally, like the portal, we do not always match the API naming as we want tp be clear on what properties do as well as provide the best ux for terraform users as many users do not use IDEs that pull in property descriptions. ie something like like cluster.ssh_setting.ssh_value has redundant ssh in it as its already has ssh in the path nor is it as clear as clear as cluster.ssh.key_value.

Reply:
Makes sense, thanks for clarifying @katbyte